homeassistant/components/tami4/__init__.py

```diff
@@ -0,0 +1,66 @@
+"""The Tami4Edge integration."""
+from __future__ import annotations
+
+from Tami4EdgeAPI import Tami4EdgeAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity import DeviceInfo, Entity, EntityDescription
+
+from .const import CONF_REFRESH_TOKEN, DOMAIN
+
+# Limit to one platform for the first PR
+# PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.BUTTON]
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class Tami4EdgeBaseEntity(Entity):
+    """Base class for Tami4Edge entities."""
+
+    def __init__(
+        self, edge: Tami4EdgeAPI, entity_description: EntityDescription
+    ) -> None:
+        """Initialize the Tami4Edge."""
+        self._state = None
+        self._edge = edge
+        self._name = f"{edge.device.name}"
+        self._device_id = edge.device.psn
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{self._device_id}_{self.entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, edge.device.psn)},
+            manufacturer="Stratuss",
+            name=edge.device.name,
+            model="Tami4",
+            sw_version=edge.device.device_firmware,
+            suggested_area="Kitchen",
+        )
```

Comment:
We generally put the base entity in a separate module, please move this into an `entity.py` file.